modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java

```diff
@@ -653,7 +669,7 @@ private void addToExpiryTable(Connection connection, AndesMessageMetadata metada
         Context contextWrite = MetricManager.timer(Level.INFO, MetricsConstants.DB_WRITE).start();
         try {
             if (metadata.getExpirationTime() > 0) {
```

Comment:
Let's not give this knowledge to the database layer. Let's have a method called as 'isExpirationSet()' which return a bollean in an appropriate class and just use it here.